src/dev_cron.erl

```diff
@@ -1,75 +1,301 @@
+every(_Msg1, Msg2, Opts) ->
+	case {
+		hb_ao:get(<<"cron-path">>, Msg2, Opts),
+		hb_ao:get(<<"every-interval">>, Msg2, Opts)
+	} of
+		{not_found, _} -> 
+			{error, <<"No cron path found in message.">>};
+		{_, not_found} ->
+			{error, <<"No every-interval found in message.">>};
+		{CronPath, IntervalString} -> 
+			try 
+				IntervalMillis = parse_time(IntervalString),
+				if IntervalMillis =< 0 ->
+					throw({error, invalid_interval_value});
+				true ->
+					ok
+				end,
+				ReqMsgID = hb_message:id(Msg2, all),
+				ModifiedMsg2 = maps:remove(<<"cron-path">>, maps:remove(<<"every-interval">>, Msg2)),
+				Pid = spawn(fun() -> every_worker_loop(CronPath, ModifiedMsg2, Opts, IntervalMillis) end),
+				Name = {<<"cron@1.0">>, ReqMsgID},
+				hb_name:register(Name, Pid),
+				{ok, ReqMsgID}
+			catch
+				error:{invalid_time_unit, Unit} -> {error, <<"Invalid time unit: ", Unit>>};
+				error:{invalid_interval_value} -> {error, <<"Invalid interval value.">>};
+				error:{Reason, _Stack} ->
+					{error, {<<"Error parsing interval">>, Reason}}
+			end
+	end.
 
--record(state, {
-    time,
-    last_run
-}).
-
-init(State = #{ <<"process">> := ProcM }, Params) ->
-    case lists:keyfind(<<"time">>, 1, Params) of
-        {<<"time">>, CronTime} ->
-            MilliSecs = parse_time(CronTime),
-            %% TODO: What's the most sensible way to initialize the last_run?
-            %% Current behavior: Timer starts after _first_ message.
-            {ok, State#{ <<"cron">> => #state { time = MilliSecs, last_run = timestamp(ProcM) } }};
-        false ->
-            {ok, State#{ <<"cron">> => inactive }}
-    end.
+%% @doc Exported function for stopping a scheduled task.
+stop(_Msg1, Msg2, Opts) ->
+	case hb_ao:get(<<"task">>, Msg2, Opts) of
+		not_found ->
+			{error, <<"No task ID found in message.">>};
+		TaskID ->
+			Name = {<<"cron@1.0">>, TaskID},
+			case hb_name:lookup(Name) of
+				Pid when is_pid(Pid) ->
+					?event({cron_stopping_task, {task_id, TaskID}, {pid, Pid}}),
+					exit(Pid, kill),
+					hb_name:unregister(Name),
+					{ok, #{<<"status">> => 200, <<"body">> => #{
+						<<"message">> => <<"Task stopped successfully">>,
+						<<"task_id">> => TaskID
+					}}};
+				undefined ->
+					{error, <<"Task not found.">>};
+				Error ->
+					?event({cron_stop_lookup_error, {task_id, TaskID}, {error, Error}}),
+					{error, #{<<"error">> => <<"Failed to lookup task or unexpected result">>, <<"details">> => Error}}
+			end
+	end.
 
+every_worker_loop(CronPath, Req, Opts, IntervalMillis) ->
+	ReqSingleton = Req#{ <<"path">> => CronPath },
+	?event({cron_every_worker_executing, {path, CronPath}, {req_id, hb_message:id(Req, all)}}),
+	try
+		Result = dev_meta:handle(Opts, ReqSingleton),
+		?event({cron_every_worker_executed, {path, CronPath}, {result, Result}})
+	catch
+		Class:Reason:Stacktrace ->
+			?event({cron_every_worker_error, {path, CronPath}, {error, Class, Reason, Stacktrace}}),
+			timer:sleep(IntervalMillis),
```

Comment:
I think these two lines might be redundant? Should fall through to the sleep below as long as you don't throw here.

Reply:
removed the sleep, but keeping the event here to have an audit of if a worker fails.